common/cf-solana/src/blake3.rs

```diff
@@ -0,0 +1,46 @@
+pub use ::blake3::Hasher;
+
+use crate::types::Hash;
+
+/// Calculates Blake3 hash of given byte slice.
+///
+/// When `solana-program` or `solana-program-2` feature is enabled and
+/// building a solana program, this is using Solana’s `sol_blake3` syscall.
+/// Otherwise, the calculation is done by `blake3` crate.
+#[allow(dead_code)]
+pub fn hash(bytes: &[u8]) -> Hash {
+    if cfg!(target_os = "solana-program") &&
+        (cfg!(feature = "solana-program") ||
+            cfg!(feature = "solana-program-2"))
+    {
+        hashv(&[bytes])
+    } else {
+        Hash(::blake3::hash(bytes).into())
```

Comment:
the blake3 is blocked on mainnet so i dont think we can use it until they enable it. Ref (https://github.com/anza-xyz/agave/wiki/Feature-Gate-Activation-Schedule#features-are-blocked)

Reply:
That is unfortunate. It’s going to be a problem when adding cf-solana to solana-ibc.